### DISCUSSION TITLE
Port numbers

### DISCUSSION THREAD

**Discussion by yunkon-kim** [cloud-barista/cloud-migrator#12] [General]
# 서브시스템에서 활용하는 포트 조사

🔈 서브시스템 및 활용하는 외부 서버들에서 활용되는 포트 번호 현황 확인을 요청 드립니다.

## Container Names and Ports in Cloud-Migrator

> [!WARNING]
> Same target port: 5432 (PostgreSQL) 

| Subsystem       | Container Name     | Target Port | Published Port | Misc.      |  Note |
| --------------- | :----------------- | ----------- | -------------- | ---------- | ----------|
| CM-Butterfly    | cm-butterfly-api   | 4000        | 4000           |            |      |
| CM-Butterfly    | cm-butterfly-front | 80          | 80             |            |      |
| CM-Butterfly    | cm-butterfly-db    | **5432**    | 543            | **PostgreSQL** |      |
| CM-Ant          | cm-ant             | 8880        | 8880           |            |      |
| CM-Ant          | cm-ant-db          | **5432**    | 5432           | **PostgreSQL** |      |
| CM-Cicada       | cm-cicada          | 8083        | 8083           |            |      |
| CM-Cicada       | airflow-mysql      | 3306        | 3306           | MySQL      |      |
| CM-Cicada       | airflow-server     | 5555, 8080  | 5555, 8080     |            |      |
| CM-Damselfly    | cm-damselfly       | 8088        | 8088           |            |      |
| CM-Beetle       | cm-beetle          | 8056        | 8056           |            |      |
| CM-Grasshopper  | cm-grasshopper     | 8084        | 8084           |            |      |
| CM-Honeybee     | cm-honeybee        | 8081        | 8081           |            |      |
| CB-Tumblebug    | cb-tumblebug       | 1323        | 1323           |            |      |
| CB-Tumblebug    | cb-mapui           | 1324        | 1324           |            |      |
| CB-Tumblebug    | cb-tumblebug-etcd               | 2379, 2380  | 2379, 2380     | ETCD       |      |
| CB-Tumblebug    | cb-tumblebug-postgres      | 6432  | 6432     | **PostgreSQL**  |  Updated on August 20, 2025 |
| CB-Spider       | cb-spider          | 1024        | 1024           |            |      |
| CM-Mayfly       | N/A                | N/A         | N/A            | N/A        |      |
| mc-data-manager | mc-data-manager    | 3300        | -              | Not in use |      |

**Comment by yunkon-kim** [comment]
각 서브시스템에서 활용하고 있는 포트를 공유해 주십시오. (+선호하는 포트가 있는 경우 함께 공유)
 
: Mention 되신 분들 중에 변경이 필요하신 분이 있으시면 코멘트로 남겨주시기 바랍니다.

@dogfootman @MZC-CSC (cc. @innodreamer)
https://github.com/cloud-barista/cm-butterfly:
* cm-butterfly-api : 4000
* cm-butterfly-front : 80
* cm-butterfly-db : 5432 (PostgreSQL)

@dev4unet @MZC-CSC (cc. @innodreamer)
https://github.com/cloud-barista/cm-mayfly:
* None

@hippo-an (cc. @seokho-son)
https://github.com/cloud-barista/cm-ant:
* 8880 (Ant API server)
* 5432 (PostgreSQL)

@ish-hcc (cc. @sykim-etri)
https://github.com/cloud-barista/cm-cicada:
* 8083 (Cicada API Server)
* 8080 (Airflow WebServer)

@innodreamer
https://github.com/cloud-barista/cm-damselfly:
* 8088 (Damselfly API Server)

@ish-hcc (cc. @sykim-etri)
https://github.com/cloud-barista/cm-grasshopper:
* 8084 (Grasshopper API Server)

@yunkon-kim @seokho-son @sykim-etri 
https://github.com/cloud-barista/cm-beetle:
* 8056 (Beetle API Server)
* (TBD) 2379, 2380 (ETCD)

@ish-hcc (cc. @powerkimhub)
https://github.com/cloud-barista/cm-honeybee:
* 8081 (Honeybee API Server)
* 8082 (Honeybee Agent)

@seokho-son @sykim-etri @yunkon-kim
https://github.com/cloud-barista/cb-tumblebug:
* 1323 (CB-TB API Server)
* 1324 (CB-MapUI)
* 2379, 2380 (ETCD)

@powerkimhub 
https://github.com/cloud-barista/cb-spider:
* 1024 (Spider Rest Server)
* 2048 (gRPC for CLI, deprecated soon)

N/A (cc. @yunkon-kim)
https://github.com/cloud-barista/mc-data-manager:
* 3300 (mc-data-manager)

**Reply by hippo-an** [reply]
cm-ant
api server: 8880
postgresql: 5432

**Comment by ish-hcc** [comment]
- cm-cicada
  - API Server: 8083
  - Airflow WebServer: 8080
- cm-honeybee
  - API Server: 8081
  - Agent: 8082
- cm-grasshopper
  - API Server: 8084

**Comment by yunkon-kim** [comment]
확인 부탁드립니다.

@dogfootman @MZC-CSC (cc. @innodreamer)
https://github.com/cloud-barista/cm-butterfly:
*

@dev4unet @MZC-CSC (cc. @innodreamer)
https://github.com/cloud-barista/cm-mayfly:
*

**Reply by dev4unet** [reply]
@yunkon-kim 
[cm-mayfly](https://github.com/cloud-barista/cm-mayfly)
- 포트를 사용하지 않습니다.

[cm-butterfly](https://github.com/cloud-barista/cm-butterfly)
- cm-butterfly-api : 4000
- cm-butterfly-front : 80
- cm-butterfly-db : 5432 (PostgreSQL)

Ant와 Butterfly 모두 DB를 PostgreSQL을 사용 중이며 둘 다 기본 포트인 5432를 이용하고 있습니다.
하나의 호스트에서 구동되는 docker-compose에서는 Host의 포트 충돌이 발생할 수 있기에 포트 영향을 받지 않도록 **target port는 5432로 고정**하고 Public 접속은 중요하지 않아서 랜덤으로 하려다 임시로 543을 설정했는데 혹시 PostgreSQL의 포트도 유니크하게 5433처럼 기본 포트를 통일할 필요는 없겠죠?
    ports:
      - target: 5432
        published: 543
        protocol: tcp

**Reply by yunkon-kim** [reply]
@dev4unet 공유 감사합니다!

Published port는 적절히 설정해 주시면 될 것 같습니다 :-) 

App의 기본 port, Published port 등에 대해 전반적으로 정리하여 본문에 올리겠습니다.

**Comment by yunkon-kim** [comment]
@dev4unet @hippo-an @ish-hcc @innodreamer @sykim-etri @seokho-son @powerkimhub 

본문의 표에 추가하거나 수정 되어야 할 부분이 있는지 확인 바랍니다.

이후, Mayfly 차원에서 Container Name, Published Port를 적절히 개선해 주시면 좋을 것 같습니다.

**Reply by ish-hcc** [reply]
@yunkon-kim 
honeybee, cicada, grasshopper 변동 사항 없습니다

**Reply by yunkon-kim** [reply]
@ish-hcc 넵! 확인 감사합니다.

**Reply by innodreamer** [reply]
@yunkon-kim Damselfly도 기입해주신 내용이 맞습니다. 감사합니다.

**Reply by yunkon-kim** [reply]
@innodreamer 넵! 감사합니다.
